From 0350336d647894e05c3967dc08d338b4700447ff Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:43 -0400
Subject: [tools] lib/traceevent: Unregister handler when cfg80211 plugin is unloaded

Message-id: <1408953210-26343-276-git-send-email-jolsa@redhat.com>
Patchwork-id: 88992
O-Subject: [PATCH RHEL7.1 BZ1133083 275/322] tools lib traceevent: Unregister handler when cfg80211 plugin is unloaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 6024cf3898d25088b01025d72a6929839de9c7b6
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 16 11:31:15 2014 +0900

description
===========
The function handler should be unregistered when the plugin is unloaded
otherwise it'll try to access invalid memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_cfg80211.c b/tools/lib/traceevent/plugin_cfg80211.c
index 57e9822..c066b25 100644
--- a/tools/lib/traceevent/plugin_cfg80211.c
+++ b/tools/lib/traceevent/plugin_cfg80211.c
@@ -22,3 +22,9 @@ int PEVENT_PLUGIN_LOADER(struct pevent *pevent)
            PEVENT_FUNC_ARG_VOID);
  return 0;
 }
+
+void PEVENT_PLUGIN_UNLOADER(struct pevent *pevent)
+{
+ pevent_unregister_print_function(pevent, process___le16_to_cpup,
+      "__le16_to_cpup");
+}
-- 
1.7.1